docs/source/pages/tutorials/npm_provenance.rst

```diff
@@ -0,0 +1,127 @@
+
+For this tutorial, we can create a policy that checks whether the three checks (as above) have passed. In this way we can be sure that the requirement is satisfied without having to dive into the reports directly.
+
+.. code-block:: c++
+
+    #include "prelude.dl"
+
+    Policy("has-verified-provenance", component_id, "Require a verified provenance file.") :-
+        check_passed(component_id, "mcn_provenance_derived_repo_1"),
+        check_passed(component_id, "mcn_provenance_derived_commit_1"),
+        check_passed(component_id, "mcn_provenance_verified_1").
+
+    apply_policy_to("has-verified-provenance", component_id) :-
+        is_component(component_id, "pkg:npm/semver@7.6.2").
+
+After including some helper rules, the above policy is defined as requiring all three of the checks to pass through the ``check_passed(<target>, <check_name>)`` mechanism. The target is then defined by the criteria applied to the policy. In this case, the artifact with a PURL that matches the version of ``semver`` used in this tutorial: ``pkg:npm/semver@7.6.2``. With this check saved to a file, say ``verified.dl``, we can run it against Macaron's database to confirm that the analysis we performed earlier in this tutorial did indeed pass all three checks.
+
+.. code-block:: shell
+
+    ./run_macaron.sh -d output/macaron.db -f verified.dl
+
+The result of this command should show that the policy we have written succeeds on the ``semver`` library. As follows:
+
+.. code-block:: javascript
+
+    component_satisfies_policy
+        ['1', 'pkg:npm/semver@7.6.2', 'has-verified-provenance']
+    component_violates_policy
+    failed_policies
+    passed_policies
+        ['has-verified-provenance']
+
+Additionally, if we had happened to run some more analyses on other versions of ``semver``, we could also apply the policy to them with only a small modification:
+
+.. code-block:: c++
+
+    apply_policy_to("has-verified-provenance", component_id) :-
+        is_component(component_id, purl),
+        match("pkg:npm/semver@.*", purl).
+
+With this modification, all versions of ``semver`` previously analysed will show up when the policy is run again. Like so:
+
+.. code-block:: javascript
+
+    component_satisfies_policy
+        ['1', 'pkg:npm/semver@7.6.2', 'has-verified-provenance']
+        ['2', 'pkg:npm/semver@7.6.0', 'has-verified-provenance']
+    component_violates_policy
+        ['3', 'pkg:npm/semver@1.0.0', 'has-verified-provenance']
+    failed_policies
+        ['has-verified-provenance']
+
+Here we can see that the newer versions, 7.6.2 and 7.6.0, passed the checks, meaning they have verified provenance. The much older version, 1.0.0, did not pass the checks, which is not surprising given that it was published 13 years before this tutorial was made!
```

Comment:
It would be good to add an exception rule next to show we can make a flexible policy. For example version above x should pass the checks only.